homeassistant/components/camera/__init__.py

```diff
@@ -107,11 +107,14 @@ class Image:
     camera = _get_camera_from_entity_id(hass, entity_id)
     camera_prefs = hass.data[DATA_CAMERA_PREFS].get(entity_id)
 
-    if not camera.stream_source:
+    with async_timeout.timeout(10):
+        source = await camera.stream_source()
+
+    if not source:
```

Comment:
Source does not always exist in this context

Reply:
I just assigned it 2 lines above ?

Reply:
If the timeout expires, an exception will be raised.